Lib/asyncio/coroutines.py

```diff
@@ -107,12 +107,14 @@ def coroutine(func):
     If the coroutine is not yielded from before it is destroyed,
     an error message is logged.
     """
-    if inspect.iscoroutinefunction(func):
+    if (inspect.iscoroutinefunction(func)
+            and not isinstance(func, functools.partial)):
```

Comment:
We don't need extra checks in this function.  Here, we check for an `async def` coroutine -- we don't need to wrap them in a `CoroWrapper` *ever*.